packages/preferences/src/browser/user-configs-preference-provider.ts

```diff
@@ -92,14 +92,24 @@ export class UserConfigsPreferenceProvider extends PreferenceProvider {
 
     async setPreference(preferenceName: string, value: any, resourceUri?: string): Promise<boolean> {
         const sectionName = preferenceName.split('.', 1)[0];
-        const configName = this.configurations.isSectionName(sectionName) ? sectionName : this.configurations.getConfigName();
-
-        const providers = this.providers.values();
-
-        for (const provider of providers) {
-            if (this.configurations.getName(provider.getConfigUri()) === configName) {
-                return provider.setPreference(preferenceName, value, resourceUri);
+        const defaultConfigName = this.configurations.getConfigName();
```

Comment:
This is also rewritten to allow us to fall back to `settings.json` if the change is rejected by section providers (e.g. `extensions.json`). There isn't any real change to the other operations.